// nuxt-snackbar/dist/module.mjs
import { defu } from 'defu';
import { defineNuxtModule, createResolver, addImports, addPlugin, addComponent } from '@nuxt/kit';

const name = "nuxt-snackbar";
const version = "1.1.2";

const module = defineNuxtModule({
  meta: {
    name,
    version,
    configKey: "snackbar",
    compatibility: {
      nuxt: ">=3.0.0"
    }
  },
  defaults: {
    top: false,
    bottom: true,
    left: false,
    right: false,
    groups: true,
    success: "#4caf50",
    error: "#ff5252",
    warning: "#fb8c00",
    info: "#2196f3",
    duration: 4e3,
    limit: 0,
    messageClass: "",
    messageActionClass: "",
    zIndex: 9999,
    dense: false,
    shadow: true,
    reverse: false,
    border: "",
    backgroundOpacity: 0.12,
    backgroundColor: "currentColor",
    baseBackgroundColor: "#ffffff",
    dismissOnActionClick: false,
    iconPresets: {}
  },
  hooks: {},
  setup(options, nuxt) {
    const resolver = createResolver(import.meta.url);
    nuxt.options.css.push("vue3-snackbar/styles");
    nuxt.options.runtimeConfig.snackbar = defu(
      nuxt.options.runtimeConfig.snackbar,
      options
    );
    nuxt.options.runtimeConfig.public.snackbar = defu(
      nuxt.options.runtimeConfig.public.snackbar,
      options
    );
    addImports({
      name: "useSnackbar",
      from: "vue3-snackbar"
    });
    addPlugin(resolver.resolve("./runtime/plugin"));
    addComponent({
      name: "NuxtSnackbar",
      filePath: resolver.resolve("./runtime/components/NuxtSnackbar")
    });
  }
});

export { module as default };
